apps/measures/models.py

```diff
@@ -197,14 +201,73 @@ def facilitative_administration(self):
 
     @property
     def measures_pages(self):
-        measures_pages = MeasuresDetailPage.objects.live().descendant_of(self)
+        measures_pages = MeasuresDetailPage.objects.live()\
+            .descendant_of(self).order_by("title", "-first_published_at")
 
         return measures_pages
 
+    def _q_filter(self, filter_list, field_name):
+        filter_params = Q()
+        for element in filter_list:
+            kwargs = {'{0}__contains'.format(field_name): element}
+            filter_params = filter_params | Q(**kwargs)
+        return filter_params
+
     def get_context(self, request):
         measures_pages = self.measures_pages
+
+        status = request.GET.get('status')
+        dis = request.GET.getlist('district')
+        reg = request.GET.getlist('reg')
+        fut = request.GET.getlist('fut')
+        inc = request.GET.getlist('inc')
+        fac = request.GET.getlist('fac')
+
+        filter_params = Q()
+
+        if status and status in dict(STATUS_CHOICES):
+            filter_params = filter_params & Q(status=status)
+
+        if dis:
+            filter_params = filter_params & self._q_filter(dis, 'districts')
+
+        if reg or fut or inc or fac:
+            fields_filter_params = Q()
+            if reg:
+                fields_filter_params = fields_filter_params | self._q_filter(
+                    reg, 'regenerative_management'
+                )
+            if fut:
+                fields_filter_params = fields_filter_params | self._q_filter(
+                    fut, 'future_opportunities_for_all'
+                )
+            if inc:
+                fields_filter_params = fields_filter_params | self._q_filter(
+                    inc, 'inclusive_shaping_of_urban_life'
+                )
+            if fac:
+                fields_filter_params = fields_filter_params | self._q_filter(
+                    fac, 'facilitative_administration'
+                )
+            filter_params = filter_params & fields_filter_params
+
+        measures_pages = measures_pages.filter(Q(filter_params))
```

Comment:
Do we really need the Q around again?

Reply:
You are right, that should be safely Q already now. :rofl: